Makefile

```diff
@@ -19,31 +19,69 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# allow optional per-repo overrides
+-include Makefile.overrides.mk
+
 RUN =
 
-# Set the enviornment variable BUILD_WITH_CONTAINER to use a container
+# Set the environment variable BUILD_WITH_CONTAINER to use a container
 # to build the repo. The only dependencies in this mode are to have make and
 # docker. If you'd rather build with a local tool chain instead, you'll need to
 # figure out all the tools you need in your environment to make that work.
+
+ifeq ($(JOB_NAME),istio.io-presubmit)
 export BUILD_WITH_CONTAINER ?= 0
+else
+export BUILD_WITH_CONTAINER ?= 1
+endif
+
 ifeq ($(BUILD_WITH_CONTAINER),1)
-IMG = gcr.io/istio-testing/build-tools:2019-08-05
+IMG = gcr.io/istio-testing/build-tools:2019-08-15
 UID = $(shell id -u)
 PWD = $(shell pwd)
 GOBIN_SOURCE ?= $(GOPATH)/bin
-GOBIN ?= /work/out/bin
+export GOBIN ?= /work/out/bin
+
+LOCAL_ARCH := $(shell uname -m)
+ifeq ($(LOCAL_ARCH),x86_64)
+GOARCH_LOCAL := amd64
+else ifeq ($(shell echo $(LOCAL_ARCH) | head -c 5),armv8)
+GOARCH_LOCAL := arm64
+else ifeq ($(shell echo $(LOCAL_ARCH) | head -c 4),armv)
+GOARCH_LOCAL := arm
+else
+GOARCH_LOCAL := $(LOCAL_ARCH)
+endif
+export GOARCH ?= $(GOARCH_LOCAL)
+
+LOCAL_OS := $(shell uname)
+ifeq ($(LOCAL_OS),Linux)
+   export GOOS_LOCAL = linux
+else ifeq ($(LOCAL_OS),Darwin)
+   export GOOS_LOCAL = darwin
+else
+   $(error "This system's OS $(LOCAL_OS) isn't recognized/supported")
+endif
+
+export GOOS ?= $(GOOS_LOCAL)
 
 RUN = docker run -t --sig-proxy=true -u $(UID) --rm \
+	-e GOOS="$(GOOS)" \
+	-e GOARCH="$(GOARCH)" \
+	-e GOBIN="$(GOBIN)" \
 	-v /etc/passwd:/etc/passwd:ro \
 	-v $(readlink /etc/localtime):/etc/localtime:ro \
+	$(CONTAINER_OPTIONS) \
 	--mount type=bind,source="$(PWD)",destination="/work" \
 	--mount type=volume,source=istio-go-mod,destination="/go/pkg/mod" \
 	--mount type=volume,source=istio-go-cache,destination="/gocache" \
 	--mount type=bind,source="$(GOBIN_SOURCE)",destination="/go/out/bin" \
 	-w /work $(IMG)
+else
+export GOBIN ?= ./out/bin
```

Comment:
I think GOBIN needs to be sanitized in common - this is an area for further refinement in common-files.